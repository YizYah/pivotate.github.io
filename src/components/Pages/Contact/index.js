import React, { Component } from "react";
import emailjs from "emailjs-com";

class Contact extends Component {
  state = {
    active: false,
  };
  render() {
    let vm = this;
    function sendEmail(e) {
      e.preventDefault();
      emailjs
        .sendForm(
          "mailgun",
          "template_qoPOGXNS",
          e.target,
          "user_PVCtRCUc6W81aPgHxjTXj"
        )
        .then(
          (result) => {
            vm.setState({
              active: !vm.state.active,
            });
          },
          (error) => {
            console.log(error.text);
          }
        );
    }

    return (
      <div
        className="container full-width-sm"
        style={{ marginBottom: "124px", marginTop: "100px" }}
      >
        <div className="row">
          <div className="contact__wrap text-center">
            <form
              disabled
              className="contact__form"
              onSubmit={sendEmail}
              style={{ lineHeight: "1.5" }}
            >
              <div>
                <h3 className="headline" style={{ marginBottom: "0" }}>
                  Get in touch with the <br /> Pivotate Team!
                </h3>
                <p
                  style={{
                    color: "#777",
                    marginBottom: "2em",
                  }}
                >
                  We're here to help you with any questions you may have.
                </p>
              </div>

              {this.state.active && (
                <div>
                  <hr />
                  Thank you for your message. Our Customer Success Team will get
                  back to you as soon as possible.
                </div>
              )}
              {!this.state.active && (
                <div>
                  <input
                    required
                    className="contact__input"
                    placeholder="Name..."
                    type="text"
                    name="name"
                  />
                  <input
                    required
                    className="contact__input"
                    placeholder="Email..."
                    type="email"
                    name="from_email"
                  />
                  <textarea
                    required
                    placeholder="Message..."
                    row="5"
                    className="contact__input"
                    name="message"
                  ></textarea>
                  <input
                    type="submit"
                    className="button button--rounded button--yellow"
                    value="SEND IT"
                    style={{
                      width: "100%",
                      minHeight: " 50px",
                      fontWeight: "600",
                      fontSize: "1em",
                    }}
                  />
                </div>
              )}
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default Contact;
