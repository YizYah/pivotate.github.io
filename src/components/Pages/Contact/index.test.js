import React from 'react';
import renderer from 'react-test-renderer';
import Contact from '../Contact';

test('Snapshopt Test', () => {
  const component = renderer.create(<Contact />);

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
