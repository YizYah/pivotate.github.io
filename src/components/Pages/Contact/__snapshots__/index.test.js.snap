// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshopt Test 1`] = `
<div
  className="container full-width-sm"
  style={
    Object {
      "marginBottom": "124px",
      "marginTop": "100px",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="contact__wrap text-center"
    >
      <form
        className="contact__form"
        disabled={true}
        onSubmit={[Function]}
        style={
          Object {
            "lineHeight": "1.5",
          }
        }
      >
        <div>
          <h3
            className="headline"
            style={
              Object {
                "marginBottom": "0",
              }
            }
          >
            Get in touch with the 
            <br />
             Pivotate Team!
          </h3>
          <p
            style={
              Object {
                "color": "#777",
                "marginBottom": "2em",
              }
            }
          >
            We're here to help you with any questions you may have.
          </p>
        </div>
        <div>
          <input
            className="contact__input"
            name="name"
            placeholder="Name..."
            required={true}
            type="text"
          />
          <input
            className="contact__input"
            name="from_email"
            placeholder="Email..."
            required={true}
            type="email"
          />
          <textarea
            className="contact__input"
            name="message"
            placeholder="Message..."
            required={true}
            row="5"
          />
          <input
            className="button button--rounded button--yellow"
            style={
              Object {
                "fontSize": "1em",
                "fontWeight": "600",
                "minHeight": " 50px",
                "width": "100%",
              }
            }
            type="submit"
            value="SEND IT"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
